modules/primo/src/Primo/BriefSearch/Client.php

```diff
@@ -40,14 +47,20 @@ class Client {
    * @param string $ipAddress
    *   The client IP. This will help to identify the institution in cases the
    *   institution was not identified (according to the IP range associated with
-   *   the institution)
+   *   the institution).
+   * @param string $scopes
+   *   Comma-seperated list of location scopes the search should be done within.
    */
-  public function __construct(ClientInterface $httpClient, $institution, $ipAddress) {
+  public function __construct(ClientInterface $httpClient, $institution, $ipAddress, $scopes = NULL) {
     $this->httpClient = $httpClient;
     $this->defaultParameters = [
       'institution' => $institution,
-      'ip' => $ipAddress
+      'ip' => $ipAddress,
     ];
+    if (!empty($scopes)) {
+      // Do some basic whitespace cleanup in case the user was sloppy.
+      $this->locationScopes = trim(str_replace(' ', '', $scopes));
```

Comment:
Accepting scopes as a string to then chop it up, feels like we are reducing the strictness of the client to make it easier to generate an admin interface.

Can't we have both?

Reply:
Im just removing accidental whitespace such as "key1, key2" (the space after the comma should not be there) and leading/trailing newlines and tabs